8-better-way-utilizing-object-pool-nocanvas-svg/js/index.js

```diff
@@ -27,22 +28,27 @@ Bubble.prototype = {
     this.vx = vx;
     this.vy = vy;
   },
-  update: function () {
-    this.x += this.vx;
-    this.y += this.vy;
-    this.vy += gravity;
-    this.vx = this.vx*friction;
-    this.vy = this.vy*friction;
+  update: function (dt) {
+    // friction opposes the direction of velocity
+    var acceleration = -Math.sign(this.vx) * friction;
+    // distance = velocity * time + 0.5 * acceleration * (time ^ 2)
+    this.x += this.vx * dt + 0.5 * acceleration * (dt ^ 2);
+    this.y += this.vy * dt + 0.5 * gravity * (dt ^ 2);
+    // velocity = velocity + acceleration * time
+    this.vy += gravity * dt;
+    this.vx += acceleration * dt;
```

Comment:
I was expecting this to take more of a toll on the CPU because there's more that we're calculating here, but did some timeline recordings and it looks like yours actually performs _better_, in part I think because of the even distribution.